multivariate_grouped_data_plot.Rmd

```diff
@@ -0,0 +1,825 @@
+ggplot(df.summary2, aes(dose, len)) +
+  geom_pointrange(
+    aes(ymin = len-sd, ymax = len+sd, color = supp),
+    position = position_dodge(0.3)
+    )+
+  scale_color_manual(values = c("#00AFBB", "#E7B800"))
+# (2) Standard error bars
+ggplot(df.summary2, aes(dose, len)) +
+  geom_errorbar(
+    aes(ymin = len-sd, ymax = len+sd, color = supp),
+    position = position_dodge(0.3), width = 0.2
+    )+
+  geom_point(aes(color = supp), position = position_dodge(0.3)) +
+  scale_color_manual(values = c("#00AFBB", "#E7B800"))
+```
+
+(3) 为多个组创建简单的线/条形图。
+
+- 线图：按组更改线型（Supp）
+
+- 条形图：按组更改填充颜色（Supp）
+
+```{r}
+# (1) Line plot + error bars
+ggplot(df.summary2, aes(dose, len)) +
+  geom_line(aes(linetype = supp, group = supp))+
+  geom_point()+
+  geom_errorbar(
+    aes(ymin = len-sd, ymax = len+sd, group = supp),
+     width = 0.2
+    )
+# (2) Bar plots + upper error bars.
+ggplot(df.summary2, aes(dose, len)) +
+  geom_bar(aes(fill = supp), stat = "identity",
+           position = position_dodge(0.8), width = 0.7)+
+  geom_errorbar(
+    aes(ymin = len, ymax = len+sd, group = supp),
+    width = 0.2, position = position_dodge(0.8)
+    )+
+  scale_fill_manual(values = c("grey80", "grey30"))
+```
+
+(4) 创建多个组的平均值+/-sd的图。 
+
+使用ggpubr软件包，该软件包将自动计算摘要统计信息并创建图形。
+
+```{r}
+library(ggpubr)
+# Create line plots of means
+ggline(ToothGrowth, x = "dose", y = "len", 
+       add = c("mean_sd", "jitter"),
+       color = "supp", palette = c("#00AFBB", "#E7B800"))
+# Create bar plots of means
+ggbarplot(ToothGrowth, x = "dose", y = "len", 
+          add = c("mean_se", "jitter"),
+          color = "supp", palette = c("#00AFBB", "#E7B800"),
+          position = position_dodge(0.8))
+```
+
+(5) 使用标准ggplot2来重现上面的线图.
+
+```{r}
+# Create line plots
+ggplot(df, aes(dose, len)) +
+  geom_jitter(
+    aes(color = supp),
+    position = position_jitter(0.2)
+    ) + 
+  geom_line(
+    aes(group = supp, color = supp),
+    data = df.summary2
+    ) +
+  geom_errorbar(
+    aes(ymin = len-sd, ymax = len+sd, color = supp),
+    data = df.summary2, width = 0.2
+    )+
+  scale_color_manual(values = c("#00AFBB", "#E7B800"))
+```
+
+
+   
+   
+   
+
+## 4 总结
+
+### 4.1 可视化已分组的连续变量的分布
+
+适用于：x轴上的分组变量和y轴上的连续变量。
+
+可能的ggplot2层包括：
+
+- geom_boxplot() 用于箱形图
+
+- geom_violin() 用于小提琴图
+
+- geom_dotplot() 用于点图
+
+- geom_jitter() 用于带状图
+
+- geom_line() 用于线图
+
+- geom_bar() 用于柱状图
+
+R代码示例：首先创建一个名为e的图，然后通过添加一个图层来完成它：
+```{r}
+ToothGrowth$dose <- as.factor(ToothGrowth$dose)
+e <- ggplot(ToothGrowth, aes(x = dose, y = len))
+```
+
+### 4.2 创建带有误差线的均值和中位数图
+
+适用于：x轴上的分组变量，y轴上的汇总连续变量（均值/中位数）。
+
+计算摘要统计信息并使用摘要数据初始化ggplot：
+```{r}
+# Summary statistics
+library(dplyr)
+df.summary <- ToothGrowth %>%
+  group_by(dose) %>%
+  summarise(
+    sd = sd(len, na.rm = TRUE),
+    len = mean(len)
+  )
+# Initialize ggplot with data
+f <- ggplot(
+  df.summary, 
+  aes(x = dose, y = len, ymin = len-sd, ymax = len+sd)
+  )
+```
+
+### 4.3 将误差线与小提琴图，点图，线形图和条形图结合起来
```

Comment:
Hashtag headers should not be followed by numbers since the hashtags will create numbered subheadings.